lib/route.js

```diff
@@ -398,11 +398,14 @@ function buildRouting (options) {
     if (closing === true) {
       /* istanbul ignore next mac, windows */
       if (req.httpVersionMajor !== 2) {
-        res.once('finish', () => req.destroy())
         res.setHeader('Connection', 'close')
       }
 
+      /* istanbul ignore else */
       if (return503OnClosing) {
```

Comment:
Can we add a `TODO` here to remove `return503OnClosing` behavior after `node@18` is `EOL`